classifiers/reference_quality/special_character_classifier/code_snippet_common.md

```diff
@@ -0,0 +1,44 @@
+```python
+import unicodedata
+from typing import List
+
+def special_character_classifier(text: str, allowed_ranges: list = None) -> bool:
+    """
+    @param text: Text to detect special characters in
+    @param allowed_ranges: List of allowed Unicode blocks as (start, end) tuples.
+    @return:  True if text contains unusual characters, False otherwise.
+    """
+    if allowed_ranges is None:
+        allowed_ranges = [
+                (0x0020, 0x007F),  # Basic Latin
+                (0x00A0, 0x00FF),  # Latin-1 Supplement
+                (0x0100, 0x017F),  # Latin Extended-A
+                (0x0180, 0x024F),  # Latin Extended-B
+                (0x2000, 0x206F),  # General Punctuation
+                (0x20A0, 0x20CF),  # Currency Symbols
+            ]
+
+    # Allowed control characters
+    allowed_controls = {"\n", "\t", "\r"}
```

Comment:
this can be a constant outside the function

Reply:
Moved it!